test/rootless.bats

```diff
@@ -8,6 +8,12 @@ PODMAN_ROOT_DATA_DIR="$BATS_TEST_DIRNAME/../build/test-storage/user"
 
 load test_helper.bash
 
+setup_file() {
+        # Workaround for rootless non systemd github.com/containers/common/pull/2042/commits/67fdd8bd3aa912efa9d8a6375c6b5d192d4f8f9a
+        #mkdir -pm1777 "$PODMAN_ROOT_DATA_DIR/../../../rootless-netns/run"
+        mkdir -pm700 "/tmp/storage-run-1000/containers/networks/rootless-netns/run"
```

Comment:
I don't think this is the right place to apply the fix [that was suggested here](https://github.com/containers/podman/discussions/22903#discussioncomment-9681335) since it needs to be applied within the docker container, not on the host.

Reply:
Although, there seems to be a mount when starting the container on the host for /tmp/storage-run-1000/containers/networks/rootless-netns/run to be at /run (presumably within the container) right now, but might've misunderstood the log. https://github.com/mgoltzsche/podman-static/actions/runs/9377299921/job/25818642965#step:5:2804

Reply:
It is not straightforward to know exactly the execution context of this – aside from trying it. I did not get a chance to look through all the changes that are on this branch yet (before allowing it to run a restricted set of actions on my fork / repo).

Reply:
Though, I can confirm applying it within the docker container makes the test pass.

Reply:
> Although, there seems to be a mount when starting the container on the host for /tmp/storage-run-1000/containers/networks/rootless-netns/run to be at /run (presumably within the container) right now, but might've misunderstood the log. https://github.com/mgoltzsche/podman-static/actions/runs/9377299921/job/25818642965#step:5:2804

Indeed, `/run` within the log message refers to the container's file system.

Reply:
I think it just [worked](https://github.com/mgoltzsche/podman-static/actions/runs/9392703091/job/25867397485#step:5:2720) – by that I mean it allowed the test to pass by indirectly following the workaround suggestion.

I may be slightly skeptical of something like this working on the first attempt. 😂

Reply:
Well I ran the e2e tests on my machine first (where it runs a lot faster) to verify that the workaround works :smile:.